src/Keboola/StorageApi/Downloader/BlobStorageRetryMiddleware.php

```diff
@@ -155,12 +166,19 @@ protected static function createRetryDecider($maxRetries)
      * Decide if the given status code indicate the request should be retried.
      *
      * @param int $statusCode Status code of the previous request.
+     * @param \Exception|null $exception
      * @param bool $isSecondary Whether the request is sent to secondary endpoint.
      *
      * @return bool            true if the request should be retried.
      */
-    protected static function generalRetryDecider($statusCode, $isSecondary)
+    protected static function generalRetryDecider($statusCode, $exception, $isSecondary)
     {
+        if ($exception instanceof \Exception &&
+            in_array(get_class($exception), self::RETRIED_EXCEPTIONS, true)
```

Comment:
Zamerne porovnavam iba danu triedu a nie podtriedy. Kedze `RequestException` je predot `ServerException` a `ClientExeption` a tie su spracovane pomocou HTTP kodov nizsie.

Reply:
Sem sa to ale nikdy nedostane, pokud je tam exception tak to udělá retry vždycky
https://github.com/keboola/storage-api-php-client/blob/master/src/Keboola/StorageApi/Downloader/BlobStorageRetryMiddleware.php#L139-L141

Reply:
@zajca vdaka, to som prehliadol, ... kedze to neloguje, tak z logov nie je jasne ci ta retry prebehla .... idem to este raz checknut.

Reply:
Jo je to to trochu debilní v tom clientovi, protože sem udělal původně vlastní lib na to, ale ta je PHP7.1+ a tady to musí běžet PHP5.6+ takže sem nakonec udělat copy paste. 
Tady to má aj test: https://github.com/keboola/php-file-storage-utils/blob/master/tests/unit/Abs/RetryMiddlewareFactoryTest.php#L62